File: lib/src/models/vote.dart
class Vote {
  final int id;
  final String user;
  final String userId;
  final String blog;
  final int voteValue;

  Vote({
    required this.id,
    required this.user,
    required this.userId,
    required this.blog,
    required this.voteValue,
  });

  factory Vote.fromJson(Map<String, dynamic> json) {
    return Vote(
      id: json['id'],
      user: json['user'],
      userId: json['user_id'],
      blog: json['blog'],
      voteValue: json['vote_value'],
    );
  }
}

File: android/app/src/main/kotlin/com/example/yogi_application/model/Vote.kt
package com.example.yogi_application.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class Vote(
    val id: Int,
    @SerialName("user_id") val userId: Int,
    val user: String,
    val comment: Int,
    @SerialName("vote_value") val voteValue: Int,
) {
    companion object {
        fun fromJson(source: String): Vote {
            return Json.decodeFromString(source)
        }
    }

    fun toJson(): String {
        return Json.encodeToString(this)
    }
}